libsolidity/ast/Types.h

```diff
@@ -1168,9 +1168,13 @@ class UserDefinedValueType: public Type
 class TupleType: public CompositeType
 {
 public:
-	explicit TupleType(std::vector<Type const*> _types = {}): m_components(std::move(_types)) {}
+	explicit TupleType(Category _category, std::vector<Type const*> _types = {})
```

Comment:
Until now, we have a one-to-one correspondence between categories and classes, so I think this should be an internal enum or bool instead.

Reply:
After thinking it over I realized that, maybe it is worth to keep InlineArrayType. Then, by choosing a data type, the compiler needs to make one decision on how to process data. And, there is no need to have any extra flag, function or do any additional checks while data processing. Category is just enough. Otherwise, we end up with `isInlineArray` function call wherever the TupleType is being processed which makes the code harder to read. Would you mind if I keep the new type?